dcos_test_utils/dcos_api_session.py

```diff
@@ -324,11 +324,22 @@ def _wait_for_srouter_slaves_endpoints(self):
                     retry_on_exception=lambda _: False)
     def _wait_for_metronome(self):
         r = self.get('/service/metronome/v1/jobs')
-        # 500 and 504 are the expected behavior of a service
+        # 404, 500 and 504 are the expected behavior of a service
         # backend that is not up and running.
-        if r.status_code == 500 or r.status_code == 504:
-            log.info("Metronome gateway timeout, continue waiting for backend...")
+        expected_error_codes = {
+            404: ('It may be the case that Admin Router is returning a 404 '
```

Comment:
Possible error codes:
503 - cache is borked, this means that Marathon or Mesos are not up yet, continue waiting
404 - metronome has not registered with Mesos yet, continue waiting
502,504 - metronome has registered with mesos, but for some reason, the leading process is not responding, we may want to treat it as an error as I am not sure what can cause such a state. We probably discuss it a bit.
500 - I do not think this should be a valid state anymore, earlier this was ARs way of saying 404 but it is fixed now.

Hope it helps.